core/src/test/java/org/apache/iceberg/variants/TestShreddedArray.java

```diff
@@ -0,0 +1,190 @@
+    assertThat(arr.get(1).asPrimitive().get()).isEqualTo("iceberg");
+    assertThat(arr.get(2)).isInstanceOf(VariantPrimitive.class);
+    assertThat(arr.get(2).asPrimitive().get()).isEqualTo(new BigDecimal("12.21"));
+  }
+
+  @Test
+  public void testSerializationMinimalBuffer() {
+    ShreddedArray arr = createShreddedArray(ELEMENTS);
+
+    VariantValue value = roundTripMinimalBuffer(arr);
+
+    assertThat(value).isInstanceOf(SerializedArray.class);
+    SerializedArray actual = (SerializedArray) value;
+
+    assertThat(actual.numElements()).isEqualTo(3);
+    assertThat(actual.get(0)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(0).asPrimitive().get()).isEqualTo(34);
+    assertThat(actual.get(1)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(1).asPrimitive().get()).isEqualTo("iceberg");
+    assertThat(actual.get(2)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(2).asPrimitive().get()).isEqualTo(new BigDecimal("12.21"));
+  }
+
+  @Test
+  public void testSerializationLargeBuffer() {
+    ShreddedArray arr = createShreddedArray(ELEMENTS);
+
+    VariantValue value = roundTripLargeBuffer(arr);
+
+    assertThat(value).isInstanceOf(SerializedArray.class);
+    SerializedArray actual = (SerializedArray) value;
+
+    assertThat(actual.numElements()).isEqualTo(3);
+    assertThat(actual.get(0)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(0).asPrimitive().get()).isEqualTo(34);
+    assertThat(actual.get(1)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(1).asPrimitive().get()).isEqualTo("iceberg");
+    assertThat(actual.get(2)).isInstanceOf(VariantPrimitive.class);
+    assertThat(actual.get(2).asPrimitive().get()).isEqualTo(new BigDecimal("12.21"));
+  }
+
+  @Test
+  public void testMultiByteElementSize() {
+    // Create large number of elements to use 4 bytes to store element size
+    List<VariantValue> elements = Lists.newArrayList();
+    for (int i = 0; i < 100_000; i += 1) {
+      elements.add(Variants.of(RandomUtil.generateString(10, random)));
+    }
+
+    List<VariantValue> data = Lists.newArrayList();
+    data.addAll(elements);
+
+    ShreddedArray shredded = createShreddedArray(data);
+    VariantValue value = roundTripLargeBuffer(shredded);
+
+    assertThat(value.type()).isEqualTo(PhysicalType.ARRAY);
+    SerializedArray arr = (SerializedArray) value;
+    assertThat(arr.numElements()).isEqualTo(100_000);
+    for (int i = 0; i < 100_000; i++) {
+      VariantTestUtil.assertEqual(arr.get(i), elements.get(i));
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {300, 70_000, 16_777_300})
+  public void testMultiByteOffsets(int len) {
+    // Use a string exceeding 255 bytes to test value offset sizes of 2, 3, and 4 bytes
+    String randomString = RandomUtil.generateString(len, random);
+    SerializedPrimitive bigString = VariantTestUtil.createString(randomString);
```

Comment:
Why not use `Variants.of(randomString)`? No need for a serialized version right?